docs/scope-and-objectives.md

```diff
@@ -0,0 +1,144 @@
+As a result, the deployment mechanisms that manage Kubernetes clusters each have unique APIs and implementations for how to handle lifecycle events like cluster creation or deletion, control plane upgrades, and node upgrades.
+
+### Goals
+
+- To manage the lifecycle (create, scale, upgrade, destroy) of Kubernetes-conformant clusters using a declarative API.
+- To work in different environments, both on-premises and in the cloud.
+- To define common operations, provide a default implementation, and provide the ability to swap out implementations for custom ones.
+- To reuse and integrate existing ecosystem components rather than duplicating their functionality (e.g. node-problem-detector, cluster autoscaler, SIG-Multi-cluster).
+- To provide a transition path for Kubernetes lifecycle products to adopt Cluster API incrementally. The result is a fragmentation of tooling, and we want to create a unified mechanism that gives users consistency around cluster management.
+
+### Non-goals
+
+- To add these APIs to Kubernetes core (kubernetes/kubernetes).
+   -  This API should live in a namespace outside the core and follow the best practices defined by api-reviewers, but is not subject to core-api constraints.
+- To manage the lifecycle of infrastructure unrelated to the running of Kubernetes-conformant clusters.
+- To force all Kubernetes lifecycle products (kops, kubespray, GKE, AKS, EKS, IKS etc.) to support or use these APIs.
+- To manage non-Cluster API provisioned Kubernetes-conformant clusters.
+- To manage a single cluster spanning multiple infrastructure providers.
+- To configure a machine at any time other than create or upgrade.
+- To duplicate functionality that exists or is coming to other tooling, e.g., updating kubelet configuration (c.f. dynamic kubelet configuration), or updating apiserver, controller-manager, scheduler configuration (c.f. component-config effort) after the cluster is deployed.
+
+## Requirements
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
+
+- Cluster API MUST provide isolation between providers.
+
+- Cluster API MUST support multiple infrastructure providers, including both on-prem and cloud providers.
+
+- Cluster API MUST be able to manage groups/sets of Kubernetes nodes that support scaling and orchestrated upgrades.
+
+- Cluster API, through a single installation, MUST be able to manage (create, scale, upgrade, destroy) clusters on multiple providers.
+
+- Cluster API SHOULD support all operating systems in scope for Kubernetes conformance.
+
+#### User Experience
+
+- Cluster API MUST be able to display the versions of Kubernetes and related components of a Node that it manages.
+
+- Cluster API MUST be able to provide sufficient information for a consumer to access a provisioned Kubernetes cluster’s API server.
```

Comment:
This statement is a bit imprecise. Is it referring to using the Kubernetes API (for example, by providing a kubeconfig file) or is it referring to access to the node that is hosting the API server process (for example, with ssh)?

Reply:
It should be the former: Kubernetes API, I do agree that's a little bit vague

Reply:
Maybe:
```
- Cluster API MUST be able to provide sufficient information for a consumer to directly access the API server of the provisioned Kubernetes cluster
```

Reply:
How about: 

```
- Cluster API MUST be able to provide sufficient information (for example, a kubeconfig file) for a consumer to directly use the Kubernetes API of the provisioned Kubernetes cluster
```

I suggested "... the the Kubernetes API", because (in my head), "direct access" to the API server still could mean ssh access or something other than the Kubernetes API.